File: Backend/ProductService/src/main/java/tn/matchmakers/productservice/service/OrderService.java
package tn.matchmakers.productservice.service;

import tn.matchmakers.productservice.dto.OrderRequestDTO;
import tn.matchmakers.productservice.dto.OrderResponseDTO;

import java.util.List;

public interface OrderService {

    OrderResponseDTO createOrder(OrderRequestDTO request);

    List<OrderResponseDTO> getOrdersByUser(String userId);

    List<OrderResponseDTO> getAllOrders();

    OrderResponseDTO getOrderById(String id);

    OrderResponseDTO cancelOrder(String id);
    OrderResponseDTO confirmDelivery(String id);
}

File: Frontend/src/app/features/products/services/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { OrderRequest, OrderResponse } from '../models/product.model';

@Injectable({ providedIn: 'root' })
export class OrderService {
  private api = 'http://localhost:8092/products/api/orders';

  constructor(private http: HttpClient) {}

  createOrder(request: OrderRequest): Observable<OrderResponse> {
    return this.http.post<OrderResponse>(this.api, request);
  }

  getOrdersByUser(userId: string): Observable<OrderResponse[]> {
    return this.http.get<OrderResponse[]>(`${this.api}/user/${userId}`);
  }

  getOrderById(id: string): Observable<OrderResponse> {
    return this.http.get<OrderResponse>(`${this.api}/${id}`);
  }

  cancelOrder(id: string): Observable<OrderResponse> {
    return this.http.patch<OrderResponse>(`${this.api}/${id}/cancel`, {});
  }
}
